motor/discovery/aws/mql_asset_objects.go

```diff
@@ -540,24 +610,38 @@ func ecrImages(m *MqlDiscovery, account string, tc *providers.Config) ([]*asset.
 	return assets, nil
 }
 
-func ec2Instances(m *MqlDiscovery, account string, tc *providers.Config) ([]*asset.Asset, error) {
+func ec2Instances(m *MqlDiscovery, account string, tc *providers.Config, whereFilter string) ([]*asset.Asset, error) {
 	assets := []*asset.Asset{}
 	type data struct {
 		Arn        string
 		Region     string
 		Tags       map[string]string
 		InstanceId string
+		State      string
+		Name       string
+		Image      map[string]interface{}
+		PublicIp   string
 	}
-	instances, err := GetList[data](m, "return aws.ec2.instances { arn instanceId tags region }")
+	query := "return aws.ec2.instances { arn instanceId tags region state publicIp image { name } }"
+	if len(whereFilter) > 0 {
+		query = fmt.Sprintf("return aws.ec2.instances.%s { arn instanceId tags region state publicIp image { name } }", whereFilter)
```

Comment:
super nit: id now have to pass the param as `where(state === 'RUNNING')`.
for readability i think it might be better to pass just `state== RUNNING` and have this as `aws.ec2.instances.where(%s)....`

Reply:
👍